components/bl-stripe-payment-component/src/helpers/use-card-element.js

```diff
@@ -0,0 +1,98 @@
+import { useState } from 'react';
+
+import { useElements, useStripe } from '../lib/react-stripe.umd.min';
+
+export function useCardElement(eventHandlers, setIsLoading, setTransactionDetails, formRef) {
+  const { onSuccessEvent, onRejectEvent, onFocusEvent, onBlurEvent, onChangeEvent } = eventHandlers;
+
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isCardValid, setIsCardValid] = useState(false);
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const isDisabled = !stripe || !elements || !isCardValid;
+
+  const handleSubmit = async event => {
+    event.preventDefault();
+
+    if (!stripe || !elements) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    const formField = event.target;
+    const billingDetails = {
+      'address_city'   : formField.city ? formField.city.value : undefined,
```

Comment:
```suggestion
      'address_city'   : formField?.city?.value,
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

And for others